src/xcode/ENA/ENA/Source/Developer Menu/Features/DMWarnOthersNotificationViewController.swift

```diff
@@ -0,0 +1,198 @@
+			time1Textfield.widthAnchor.constraint(equalToConstant: 60),
+			time2Textfield.widthAnchor.constraint(equalToConstant: 60)
+		])
+		
+		// MARK: Set Default Value for the notification text filelds:
+		time1Textfield.text = "\(warnOthers.notificationOneTimer)"
+		time2Textfield.text = "\(warnOthers.notificationTwoTimer)"
+		
+		//Set the keyboard type.
+		time1Textfield.keyboardType = .numberPad
+		time2Textfield.keyboardType = .numberPad
+		self.hideKeyboardWhenTappedAround()
+		
+	}
+	
+	// MARK: Private API
+	@objc
+	private func scheduleNotificationsButtonTapped() {
+		let time1 = Int(time1Textfield.text ?? "")
+		let time2 = Int(time2Textfield.text ?? "")
+		
+		// Create an alert when user enter the wrong values.
+		if time2 ?? WarnOthersNotificationsTimer.timerOneTime.rawValue <= time1 ?? WarnOthersNotificationsTimer.timerTwoTime.rawValue {
+			
+			// Display second notification should be greater than first notification alert.
+			alertMessage(titleStr: "Please Enter the Correct Notification Time", messageStr: "Second notification time seconds should be greater than the first notification time seconds.")
+
+		} else {
+			// Save the notifications time into the SecureStore.
+			warnOthers.notificationOneTimer = TimeInterval(time1 ?? WarnOthersNotificationsTimer.timerOneTime.rawValue)
+			warnOthers.notificationTwoTimer = TimeInterval(time2 ?? WarnOthersNotificationsTimer.timerTwoTime.rawValue)
+			
+			//Display notification save alert.
+			alertMessage(titleStr: "Notifications time saved", messageStr: "Notification1 time \(time1 ?? WarnOthersNotificationsTimer.timerOneTime.rawValue) seconds & Notification2 time \(time2 ?? WarnOthersNotificationsTimer.timerTwoTime.rawValue) seconds has saved into the secure store.")
+		}
+
+	}
+	
+	
+	@objc
+	private func resetDefaultsButtonTapped() {
+		time1Textfield.text = "\(WarnOthersNotificationsTimer.timerOneTime.rawValue)"
+		time2Textfield.text = "\(WarnOthersNotificationsTimer.timerTwoTime.rawValue)"
+	}
+	
+	@objc
+	private func resetNotificationsButtonTapped() {
+		warnOthers.reset()
+		alertMessage(titleStr: "Done", messageStr: "Warn others notifications can appear again.")
+
+	}
+}
+
+// MARK: Set the alert popups.
+extension DMWarnOthersNotificationViewController {
+	func alertMessage(titleStr: String, messageStr: String) {
+		// create the alert
+		let alert = UIAlertController(title: titleStr, message: messageStr, preferredStyle: UIAlertController.Style.alert)
+		// add an action button
+		alert.addAction(UIAlertAction(title: "OK", style: UIAlertAction.Style.default, handler: nil))
+		// show the alert
+		self.present(alert, animated: true, completion: nil)
+	}
+}
+
+// MARK: Dismiss the keyboard
+extension DMWarnOthersNotificationViewController {
+	func hideKeyboardWhenTappedAround() {
+		let tap: UITapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(DMWarnOthersNotificationViewController.dismissKeyboard))
```

Comment:
```suggestion
		let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
```

You can use type inference  here 🤓

Reply:
Done